### PR TITLE
Add support for WEC-2103 temperature/humidity sensor

### PR DESCRIPTION
**Pull request by tthurnreiter**
This temperature/humidity transmitter came with a cheap weather station.

The only available information on it that I could find is here: https://fcc.report/FCC-ID/WEC-2103

### CODE REVIEW

File: src/devices/wec2103.c [hunk 1]
```diff
@@ -0,0 +1,95 @@
+/** @file
+    WEC-2103 temperature/humidity sensor
```
Comment (zuckschwerdt) [inline src/devices/wec2103.c:2]: Should end with a dot "." and be repeated as first line in the next doc-comment.

File: src/devices/wec2103.c [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
```
Comment (zuckschwerdt) [inline src/devices/wec2103.c:3]: Add some author copyright please.

File: src/devices/wec2103.c [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+ */
+
+/*
+-X 'n=name,m=OOK_PPM,s=2000,l=4000,g=4400,r=9400'
+
+Similar to prologue, kedsum, esperanza_ews, s3318p
+Only available information for this device: https://fcc.report/FCC-ID/WEC-2103
+
+Example datagram:
+ f2 90          6b5        96       1     8
```
Comment (zuckschwerdt) [inline src/devices/wec2103.c:17]: Needs to be valid markdown, i.e. plain blocks indented by four spaces, lists with "- " prefix

File: src/devices/wec2103.c [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    int temp_raw  = (b[2] << 4) | ((b[3] & 0xf0) >> 4);
+
+    int device_id = b[0];
+    int channel   = b[4] & 0x0f;
+    int flags = b[1] & 0xf;
+    float temp_f  = (temp_raw - 900) * 0.1f;
+    int humidity  = ((b[3] & 0x0f) * 10) + ((b[4] & 0xf0) >> 4);
+    int button   = (b[1] & 0x08) >> 3;
+
+    data = data_make(
```
Comment (zuckschwerdt) [inline src/devices/wec2103.c:59]: Surround with `/* clang-format off */` `/* clang-format on */` as seen in other decoders.

File: src/devices/wec2103.c [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+            "button",           "Button",       DATA_INT,    button,
+            //"mic",              "Integrity",    DATA_STRING, "CRC",
+            NULL);
+
+    decoder_output_data(decoder, data);
+    return 1;
+}
+
+static char *output_fields[] = {
+    "model",
```
Comment (zuckschwerdt) [inline src/devices/wec2103.c:76]: Continued line indent is 8 spaces.

File: src/devices/wec2103.c [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+r_device wec2103 = {
+    .name           = "WEC-2103 temperature/humidity sensor",
+    .modulation     = OOK_PULSE_PPM,
+    .short_width    = 1900,
+    .long_width     = 3800,
+    .gap_limit      = 4400,
+    .reset_limit    = 9400,
+    .decode_fn      = &wec2103_callback,
+    .disabled       = 0,
```
Comment (zuckschwerdt) [inline src/devices/wec2103.c:93]: remove `disabled` line

File: src/devices/wec2103.c [hunk 7]
```diff
@@ -0,0 +1,93 @@
+/** @file
+    WEC-2103 temperature/humidity sensor
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+ */
+
+/*
```
Comment (zuckschwerdt) [inline src/devices/wec2103.c:10]: Decoder doc-comment needs to start with `/**`

**Comment by zuckschwerdt** [comment]
Looks good, thanks! Isn't it more likely that a checksum would be in the last bits at the end? Do you have som example  raw codes?

**Comment by tthurnreiter** [comment]
The checksum is just speculation, I wrote this a year ago, so the details are a bit hazy. There's also an unknown byte at the end, but IIRC that one is constant across different messages.

**Comment by tthurnreiter** [comment]
[g007_433.92M_250k.zip](https://github.com/merbanan/rtl_433/files/9639295/g007_433.92M_250k.zip)

Here is some raw recordings.

**Pull request merged**